disperser/common/blobstore/shared_storage.go

```diff
@@ -90,6 +92,12 @@ func (s *SharedBlobStore) StoreBlob(ctx context.Context, blob *core.Blob, reques
 	metadataKey.BlobHash = blobHash
 	metadataKey.MetadataHash = metadataHash
 
+	refreshedMetadata, err := s.GetBlobMetadata(ctx, metadataKey)
+	if !errors.Is(err, disperser.ErrMetadataNotFound) {
+		s.logger.Error("error intentionally get blob status before creation", "err", err, "status", refreshedMetadata.BlobStatus, "metadataKey", refreshedMetadata.GetBlobKey().String())
+		return metadataKey, errProcessingInitialization
```

Comment:
what is the gRPC error code that will be returned to end user from here?

Reply:
this would be this

api.NewInternalError("failed to store blob, please try again later")

Reply:
This seems good point: How would users store the same metadataKey twice? Even if sending the same blob, it'll result in different metadataKey, so there is no overwrite.